### PR TITLE
allow val def at start of lambda

### LINKED ISSUE #47: val declaration parsed as postfix expression at start of anonymous lambda
notice the `val` is not highlighted
image

Here's what tree-sitter thinks the code is
image

notice that it thinks that the `left`-hand side of the `assignment_expression` is a `postfix_expression` instead of that the entire `assignment_expression` is actually just a `val_definition`.

This issue appears to go away if I simply precede the `val_definition` with any line of code
image

However, even though the highlighting is correct, tree-sitter no longer thinks the now-correctly-identified `val_definition` _is even within the `lambda_expression` at all!_
image

### PR DESCRIPTION
**Pull request by stumash**
resolves #47

### CODE REVIEW

File: grammar.js
Comment (stumash) [inline grammar.js:615]: this is the one of two meaningful changes in this PR, the rest is just to make it compile

File: grammar.js
Comment (stumash) [inline grammar.js:73]: this is the second and final _meaningful change_ in this PR, the rest is just to make it compile

**Comment by stumash** [comment]
apologies in advance, I never know who to tag to get these PRs reviewed

@ckipp01 @stevanmilic @maxbrunsfeld 

while we're all here, can one of you tell me if you're all the right people to tag? would like to bother the smallest necessary group of people going forward.

thanks again for the review, and happy tree-sittin'

**Comment by ckipp01** [comment]
> apologies in advance, I never know who to tag to get these PRs reviewed

Hey @stumash! @maxbrunsfeld is the only one with merge permissions on the repo I believe.

**Comment by aryx** [comment]
There's CI failures to fix though before being able to merge this.

**Comment by maxbrunsfeld** [comment]
@aryx I added r2c as maintainers of this repository too.

**Comment by aryx** [comment]
thx

**Comment by stumash** [comment]
so is this PR blocked on getting the windows CI build fixed? the last PR was merged despite the windows CI issue.

**Comment by sideeffffect** [comment]
> I added r2c as maintainers of this repository too.

as in @returntocorp?

**Review by aryx — APPROVED** [review]

**Comment by aryx** [comment]
The windows failure seems unrelated (I think we have the same issue in a few other tree-sitter repos), so I'll merge this.

**Pull request merged**
